sdk/core/azure-core-amqp/test/ut/claim_based_security_tests.cpp

```diff
@@ -71,46 +87,61 @@ namespace Azure { namespace Core { namespace Amqp { namespace Tests {
 
       GTEST_LOG_(INFO) << "CbsOpens" << static_cast<CbsOpenResult>(32768);
     }
+
+#if ENABLE_RUST_AMQP
+    session.End({});
+    connection.Close({});
+#endif
   }
 #endif // !defined(AZ_PLATFORM_MAC)
 
 #if !defined(AZ_PLATFORM_MAC)
+#if ENABLE_UAMQP
+  // Rust AMQP will fail at the connection level if there is no listener, so this particular test
+  // isn't very interesting.
   TEST_F(TestCbs, CbsOpenNoListener)
   {
-#if ENABLE_UAMQP
-    MessageTests::AmqpServerMock mockServer;
     ConnectionOptions options;
+    MessageTests::AmqpServerMock mockServer;
     options.EnableTrace = true;
     options.Port = mockServer.GetPort();
     Connection connection("localhost", nullptr, options);
     Session session{connection.CreateSession()};
     {
       ClaimsBasedSecurity cbs(session);
-      GTEST_LOG_(INFO) << "Expected failure for Open because no listener." << mockServer.GetPort();
+      GTEST_LOG_(INFO) << "Expected failure for Open because no listener.";
 
       EXPECT_EQ(CbsOpenResult::Error, cbs.Open());
     }
-#else
-    EXPECT_TRUE(false);
-#endif
   }
+#endif
 
   TEST_F(TestCbs, CbsOpen)
   {
+    ConnectionOptions options;
 #if ENABLE_UAMQP
     MessageTests::AmqpServerMock mockServer;
 
     mockServer.EnableTrace(false);
 
-    ConnectionOptions options;
     options.Port = mockServer.GetPort();
+#elif ENABLE_RUST_AMQP
+    options.Port = 25672;
+#endif
     options.EnableTrace = true;
     options.ContainerId = testing::UnitTest::GetInstance()->current_test_info()->test_case_name();
     Connection connection("localhost", nullptr, options);
+#if ENABLE_RUST_AMQP
```

Comment:
conditional testing... could we break it into a separate test that runs only if RUST AMQP ?

Reply:
Oh, that's a *REALLY* good thought.

I was trying to ensure that the tests were mostly in common between the uAMQP tests and the Rust AMQP tests. 

I *think* I can achieve the same with an appropriate level of subroutine. And it will simplify a lot of the test boiler plate.